pkg/controller/registry/reconciler/configmap.go

```diff
@@ -513,5 +515,23 @@ func (c *ConfigMapRegistryReconciler) CheckRegistryServer(logger *logrus.Entry,
 	}
 
 	healthy = true
+	var forceDeletionErrs []error
+	for _, pod := range pods {
+		if !isPodDead(pod) {
+			logger.WithFields(logrus.Fields{"pod.namespace": source.GetNamespace(), "pod.name": pod.GetName()}).Debug("pod is alive")
+			continue
+		}
+		healthy = false
+		forceDeletionErrs = append(forceDeletionErrs, pkgerrors.Errorf("found %s in a deleted but not removed state", pod.Name))
```

Comment:
Do we need to return an error from this function when we find a wedged pod and successfully delete it? My initial thought is that we would only error from this function if:
- we failed to determine healthy
- we failed to delete wedged pods.

Reply:
I started out thinking the same, but then realized that that'd be an "artificial" error. If we include both of the scenarios you mentioned in the definition of "error", we're sort of sullying the definition which is "something went wrong". In the case of "we failed to determine healthy", we are already sending that signal through the boolean variable. So decided to not include both, and only include the second scenario of "we failed to delete the wedged pods".